plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ColumnWithAliasFormatter.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.spi.connector.ConnectorSession;
+
+import static com.google.common.base.Splitter.fixedLength;
+import static com.google.common.base.Strings.padStart;
+
+public class ColumnWithAliasFormatter
+{
+    public static final int DEFAULT_COLUMN_ALIAS_LENGTH = 30;
+    public static final int ORIGINAL_COLUMN_NAME_LENGTH = 24;
+
+    public JdbcColumnHandle format(ConnectorSession session, JdbcColumnHandle column, int nextSyntheticColumnId)
+    {
+        int sequentialNumberLength = DEFAULT_COLUMN_ALIAS_LENGTH - ORIGINAL_COLUMN_NAME_LENGTH - 1;
+
+        String originalColumnNameTruncated = fixedLength(ORIGINAL_COLUMN_NAME_LENGTH)
+                .split(column.getColumnName())
+                .iterator()
+                .next();
+        String formatString = "%s_%0" + sequentialNumberLength + "d";
+        String columnName = originalColumnNameTruncated + "_" + padStart(Integer.toString(nextSyntheticColumnId), sequentialNumberLength, '0');
```

Comment:
why do we pad the next synthetic id part, it leads to identifiers with long numbers in them.
i.e. it leads to identifiers like `my_col_000001` instead of `my_col_1`.

Is this required?

Reply:
Also what happens when the `nextSyntheticColumnId` is "large" and causes the generated column name to exceed `DEFAULT_COLUMN_ALIAS_LENGTH`?

In other words what prevents the generated column name from exceeding `DEFAULT_COLUMN_ALIAS_LENGTH` and failing like it does already today?

Reply:
> why do we pad the next synthetic id part, it leads to identifiers with long numbers in them. i.e. it leads to identifiers like `my_col_000001` instead of `my_col_1`.
> 
> Is this required?

No, it's not required. I was trying to improvise here and my thinking was that maybe leading zeros make it easier to read. I was also trying to be transparent with the proposal and documented this proposal here:

https://starburstdata.atlassian.net/wiki/spaces/STARBURST/pages/3016261906/WiP+Oracle+Connector+pushdown+appends+column+names+so+that+they+exceed+maximum+identifier+length+and+fail+the+query#Possible-Solutions

and also asked for a review of the document here: https://starburstdata.atlassian.net/browse/SEP-11926 (remember I'm a board visualisation freak).

I'm  only mentioning that in order to try to align with the framework (as we discussed, we already agree wrt principles). Where I come from we used to first document like I did above, so that people can have a 'preview' of the proposed solution as soon as possible. If this process looks different here, then let me know and I'll stop creating unnecessary overhead before implementing. 

To summarize, if you think it does not help to have the padding and it obscures the query, I'm fine with dropping leading zeros. I don't have any other experience with reading the query in the debugger other than fixing this bug, so I really cannot decide by myself which way is better. I'm relying on your opinion here.

Reply:
Principle of least surprise is what I'm following here - not padding makes for least surprise since if the identifier fits the limit already then nothing will change.

Reply:
- removed padding
- added parametrized test for max integer
- added test for negative sequential number